### DISCUSSION TITLE
Access encoded JWT from Session instance

### DISCUSSION THREAD

**Discussion by LeoBorai** [nextauthjs/next-auth#2974] [Help] [answered]
When implementing callbacks I'm wondering if access to the encoded `JWT` would be possible.
As of now I understand that the session callback exposes `session`, `user` and `token` (the decoded one), but
I'm wondering if access to the encoded token is possible.

Something like:

```ts
    async session({ session, user, token, encodedToken }): Promise<Session> {
        // -- snip --
    }
```

My goal is to attach extra data to my `Session` which relies on using the JWT signed by NextAuth. So instead of having to retrieve such JWT manually I could take it straight from the `Session` instance when using either `useSession` or `getSession`.

**Comment by balazsorban44** [comment, marked as answer]
we don't expose the raw token to the frontend, but you can read it on the backend using `getToken({ raw: true })`

https://next-auth.js.org/configuration/options#jwt-helper

Also, you cannot modify the session directly from the client, since the session is saved in an HttpOnly cookie.

There is a discussion on #2269 to make updating the session easier

**Reply by LeoBorai** [reply]
Thanks!
